jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/io/FrameFlusher.java

```diff
@@ -52,6 +53,8 @@
     private final Deque<FrameEntry> queue = new ArrayDeque<>();
     private final List<FrameEntry> entries;
     private final List<ByteBuffer> buffers;
+    private final LongAdder messagesOut = new LongAdder();
+    private final LongAdder bytesOut = new LongAdder();
```

Comment:
In the highly unlikely event that we exceed the `signed long` size, what happens?
Just bad data? or a crash?

Reply:
just tested this, it will just overflow to its lowest value, so no error just bad data

Reply:
but i don't think that is even possible, that would be over 9 million terabytes transferred

Reply:
Considering that a single WebSocket frame (per the spec) can be 9,223,372,036,854,775,807 bytes (~= 9.22 exabytes) I don't think its too outlandish to think about these kinds of things.
Heck, the Servlet spec used to have an int for Content-Length [`getContentLength()`](https://docs.oracle.com/javaee/7/api/javax/servlet/ServletRequest.html#getContentLength--).
They had to add [`getContentLengthLong()`](https://docs.oracle.com/javaee/7/api/javax/servlet/ServletRequest.html#getContentLengthLong--) in Servlet 3.1 because that wasn't long enough.

Reply:
Yes, `int` is not enough, but `long` is enough. We don't want to use `BigInteger` here.

Reply:
we do longs for http, so I think this is fine

Reply:
I'm not advocating a change from `LongAdder`, just want to know what happens when it hits the end.
Options were ...

* It overflows and goes into negative values - result is bad data
* It overflows and goes back to 0 - result is bad data
* It crashes - results are unexpected
* It stops at MAX_LONG and goes no further - results in missing data (this is what the Servlet Spec does btw)